src/env.ts

```diff
@@ -0,0 +1,23 @@
+import { getCurrent } from "./component/component_node";
+
+export interface Env {
+  [key: string]: any;
+}
+
+/**
+ * This hook is useful as a building block for some customized hooks, that may
+ * need a reference to the env of the component calling them.
+ */
+export function useEnv<E extends Env>(): E {
+  return getCurrent()!.component.env as any;
+}
+
+/**
+ * This hook is a simple way to let components use a sub environment.  Note that
+ * like for all hooks, it is important that this is only called in the
+ * constructor method.
+ */
+export function useSubEnv(nextEnv: Env) {
```

Comment:
it's not clear that `nextEnv` is just a partial information that will be added to the env